File: project/league_management/src/main/java/com/iforddow/league_management/jpa/entity/team/Player.java
package com.iforddow.league_management.jpa.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "player")
public class Player {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "player_id", nullable = false)
    private Integer id;

    @Column(name = "name", length = 50)
    private String name;

    @Column(name = "height", length = 10)
    private String height;

    @Column(name = "weight", length = 10)
    private String weight;

    @Column(name = "skill_level", length = 50)
    private String skillLevel;

    @Column(name = "dob")
    private LocalDate dob;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;

}

File: project/league_management_frontend/src/lib/types/league/player.ts
/* 
A TypeScript interface representing a player in a league.

@Author: IFD
@Date: 2025-04-01
*/
export interface Player {
    playerId: number;
    name: string;
    height: number; // in inches
    weight: number; // in lbs
    skillLevel: string;
    dob: Date; // date of birth
    teamId: number; // team ID
}